app/assets/javascripts/hyrax/autocomplete.es6

```diff
@@ -1,66 +1,47 @@
-export default class Autocomplete {
-  constructor() {
-  }
-
-  // This is the initial setup for the form.
-  setup() {
-      $('[data-autocomplete]').each((index, value) => {
-          let selector = $(value)
-          switch (selector.data('autocomplete')) {
-            case "subject":
-              this.autocompleteSubject(selector);
-              break;
-            case "language":
-              this.autocompleteLanguage(selector);
-              break;
-            case "based_near":
-              this.autocompleteLocation(selector);
-              break;
-            case "work":
-              var user = selector.data('user');
-              var id = selector.data('id');
-              this.autocompleteWork(selector, user, id);
-              break;
-          }
-      });
-  }
-
-  // attach an auto complete based on the field
-  fieldAdded(cloneElem) {
-    var $cloneElem = $(cloneElem);
-    // FIXME this code (comparing the id) depends on a bug. Each input has an id and
-    // the id is duplicated when you press the plus button. This is not valid html.
-    if (/_based_near$/.test($cloneElem.attr("id"))) {
-        this.autocompleteLocation($cloneElem);
-    } else if (/_language$/.test($cloneElem.attr("id"))) {
-        this.autocompleteLanguage($cloneElem);
-    } else if (/_subject$/.test($cloneElem.attr("id"))) {
-        this.autocompleteSubject($cloneElem);
+export class Autocomplete {
+    constructor(options) {
+	console.log("Autocompleting...");
+	this.autocompleteFields = options.autocompleteFields;
+    }
+    // This is the initial setup for the form.
+    setup() {
+	$('[data-autocomplete]').each((index, value) => {
+            let selector = $(value);
+	    let autocompleteData = selector.data('autocomplete');
+	    this.activateFields(autocompleteData,selector);
+	});
+    }
+    // This activates autocomplete for added fields 
+    fieldAdded(cloneElem) {
+	let selector = $(cloneElem);
+	let autocompleteData = selector.data('autocomplete');
+	this.activateFields(autocompleteData,selector);
+    }
+    autocomplete(field) {
+	let fieldName = field.data('autocomplete');
+	let autocomplete = require('hyrax/autocomplete/default');
+	switch (fieldName) {
+	case "work":
+	    let work = require('hyrax/autocomplete/work');
```

Comment:
instead of using require, do `import Work from './work'` at the top of the file.